sources/presentation/Stride.Core.Quantum/References/ObjectReference.cs

```diff
@@ -134,9 +134,6 @@ internal IObjectNode SetTarget(IObjectNode targetNode)
                 if (targetValue != null && !type.IsInstanceOfType(targetValue))
                     throw new InvalidOperationException(@"The type of the retrieved node content does not match the type of this reference");
 
-                if (targetValue != null && !type.IsInstanceOfType(targetValue))
-                    throw new InvalidOperationException("TargetNode type does not match the reference type.");
-
```

Comment:
I removed this line entirely instead of implementing the suggested change as testing `targetNode` against the type doesn't make much sense in that context.